### ISSUE TITLE
Parameterized classes are incorrectly included into test targets that filter them out

### ISSUE THREAD

**Issue by valeraz** [Flank/flank#1242]
**Describe the bug**
> A clear and concise description of what the bug is.

Test classes annotated with `@RunWith(Parameterized::class)` end up running as part of flank invocations that do not include these classes in the config.

**To Reproduce**
> Steps to reproduce the behavior:

1. Create an parameterized instrumentation test class in package `foo.bar`. For example:
```
package foo.bar 

@RunWith(Parameterized::class)
class FooTest(private val horizontal: Boolean) {

  @Test
     test_one() {
       assert(true)
  }

    private companion object {
    @JvmStatic
    @Parameterized.Parameters
    fun data(): Collection<Boolean> = listOf(
      true,
      false
    )
  }
}
```

2. configure flank to run tests in a different package. For example:
```
test_targets:
- package foo.baz
```

**Expected behavior**
> A clear and concise description of what you expected to happen.

parameterized class from step 1 should not be executed as part of the flank run

Actual: see see something like this:
80 tests + 1 parameterized class / 81 shards (where the parameterized class is outside of the configured target)

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot? No
>
> Post the output of ` flank --version`.
version: v20.09.3

**Additional context**
> Add any other context about the problem here.

**Comment by pawelpasterz** [comment]
I think this is the same issue as https://github.com/Flank/flank/issues/1234, fixed in https://github.com/Flank/flank/pull/1233
I was unable to reproduce this error with the current master `git checkout d7b74e` but this error occurred without #1233 fix (`git checkout f34ee09` commit before fix)

@valeraz if it's not a problem, let me know if you receive the same error with a snapshot version

**Comment by pawelpasterz** [comment]
Closing due to the impossibility of reproducing.
